src/lib/ext-activitylog.js

```diff
@@ -165,6 +169,23 @@ class Controller {
     this.model[isFilterRemoved ? 'removeFilter' : 'addFilter'](filterDetail);
     this.view.setLogFilter((log) => this.isFilterMatched(log));
   }
+
+  onKeywordChange(newKeyword) {
+    this.model.setFilterKeyword(newKeyword);
+    this.view.setLogFilter((log) => this.isFilterMatched(log));
+  }
+
+  isFilterMatched(log) {
+    for (const key of Object.keys(this.model.filter)) {
+      if (this.model.filter[key].includes(log[key])) {
+        return true;
+      } else if (key === 'keyword') {
```

Comment:
So now you're looping over all (fixed set of) keys of the filter, and doing conditionals on some keys.

This is not really effective, nor readable. I suggest to put the filter logic in the model, and unroll the loop (i.e. explicitly define the filter logic for every filter key).

Reply:
done :-)